src/security/safe_command/api.py

```diff
@@ -19,64 +33,596 @@
     )
 )
 
-BANNED_EXECUTABLES = frozenset(("nc", "curl", "wget", "dpkg", "rpm"))
+BANNED_EXECUTABLES = frozenset(
+    ("nc", "netcat", "ncat", "curl", "wget", "dpkg", "rpm"))
+BANNED_PATHTYPES = frozenset(
+    ("mount", "symlink", "block_device", "char_device", "fifo", "socket"))
+BANNED_OWNERS = frozenset(("root", "admin", "wheel", "sudo"))
+BANNED_GROUPS = frozenset(("root", "admin", "wheel", "sudo"))
+BANNED_COMMAND_CHAINING_SEPARATORS = frozenset(("&", ";", "|", "\n"))
+BANNED_COMMAND_AND_PROCESS_SUBSTITUTION_OPERATORS = frozenset(
+    ("$(", "`", "<(", ">("))
+BANNED_COMMAND_CHAINING_EXECUTABLES = frozenset((
```

Comment:
The list of `BANNED_COMMAND_CHAINING_EXECUTABLES` is from a combination of lists provided by the SANS organization + a list of common shells/interpreters + a few suggestions from chatGPT + a few other binaries I know can be used for this purpose because I have used them myself. For example `byobu-ugraph` is my favorite Azure WAF workaround. 

`"script", "scriptreplay", "scriptlive"` were suggested by GPT, and I have not verified them yet so they can probably be removed.